File: compress.py
import numpy as np

def compress(w,r):
    if (r < 0) or (r > 1):
        raise Exception("R must be between zero and one")
    N = len(w)
    Nr = np.floor(N*r)
    ww = np.sort(np.abs(w))
    tol = np.abs(ww[Nr])
    
    for i in range(N):
        if np.abs(w[i]) < tol:
             w[i] = 0  
    wc = w
    return wc
    

File: compress.m
function wc=compress(w,r) 
if (r < 0) ||(r>1 )
    error('r should be between 0 and 1')
end
N=length(w); Nr=floor(N*r);
ww=sort(abs(w));
tol=abs(ww(Nr));
for j=1:N
    if (abs(w(j)) < tol)
        w(j)=0;
    end
end
wc=w;
